### ISSUE TITLE
Pipeline failing without providing console output

### ISSUE THREAD

**Issue by omnidavesz** [gocd/gocd#1996]
##### Issue Type
- Bug Report
##### Summary

Automated pipeline/stage (i.e. On Success = true) fails and doesn't produce a console output
##### Environment

Can't find this.
###### Basic environment details

Go Version: Server version 15.2.0

JAVA Version: 8

OS: Ubuntu 15.10 running in a Docker container
##### Steps to Reproduce

Can't do this as specific to our pipelines
##### Expected Results

I've generated a new pipeline group which has Build pipeline made up of 2 stages and two deploy pipelines (one for our DEV environment and one for our PROD environment). What should occur is that when I make a commit to Git the Build pipeline kicks off and both stages complete. After the second stage of the Build pipeline completes successfully the first stage of the DEV deploy pipeline should begin.
##### Actual Results

I commit to Git, the Build pipeline kicks off and completes successfully. The first stage of the DEV deploy pipeline then starts but immediately fails. No console output is produced and no information is provided in the UI. The first stage consists of 1 job with 1 task. The task is a Fetch task which is used to get an artifact generated by the Build pipeline. I've attempted to add a stage before this consisting of a single job and task which just outputs "Starting" via a terminal echo command but that also fails.
##### Log snippets

This is the issue

**Comment by zabil** [comment]
@omnidavesz can provide log snippets from go-server.log
Here's the location https://www.go.cd/documentation/user/current/installation/install/server/linux.html#location-of-gocd-server-files

**Comment by omnidavesz** [comment]
Hi,

I have had a look at those logs already but I couldn't find anything there. I understand that I haven't provided very much information but if I had more I'd be able to investigate it. Perhaps someone who has experienced a similar issue would be able to weigh in.

Cheers

**Comment by jyotisingh** [comment]
Does this happen all the time for this specific pipeline? If so could you provide the pipeline config xml.
If not, do you remember changing the material to something else after the pipeline was triggered?

**Comment by omnidavesz** [comment]
Hi Jyoti,

Attached below you'll find a zip file containing the config.xml for the pipeline group and the templates it depends on (with all sensitive information removed of course). Unfortunately I can't provide you with logs as none appear to be generated for the failing pipeline (PS_h-world-lambda-lambda-Java_DEV). And yes, it happens all the time for this pipeline. 

[GoCDXML.zip](https://github.com/gocd/gocd/files/152552/GoCDXML.zip)

Regards,
David

**Comment by omnidavesz** [comment]
Hello,

Once again sorry to chase on this issue but I was wondering if there was any update on this issue?

Regards,
Davod

**Comment by jyotisingh** [comment]
@omnidavesz - I did try to reproduce this with the config you provided but no luck there. I could be wrong, but it could possibly be because of values of one of the fields( that have been masked out in the xml you sent). Could you start your go server with debug level logging enabled to see if something shows up in logs? You would have to update /etc/go/log4j.properties :
```
log4j.logger.com.thoughtworks.go=INFO
```
BTW, when you say nothing shows up in logs, do you mean no error shows up or nothing seems to be getting logged in general for other usages as well?

**Comment by omnidavesz** [comment]
Hello again,

So it turns out that this is a one off issue with the pipeline. I've had other users generate pipelines with the same template and they have not had any issues regarding the pipeline failing at the deploy step.

I'll update the logger and try to see what is generated. And when I mentioned that no errors show up I mean that the console on the Job page on the web UI shows no output of any kind.

Regards,
David

**Comment by jyotisingh** [comment]
> And when I mentioned that no errors show up I mean that the console on the Job page on the web UI shows no output of any kind.

Ok. When this happens again please check the file located on Go-server at `/var/log/go-server/go-server.log`. The error is being generated on the server side so nothing will show up on `console.log` which contains logs from the agent machine. I am assuming for such failed builds you do not see any agent being assigned to the job as well. 
Have you tried to force trigger this pipeline and if so does that work?

**Comment by omnidavesz** [comment]
OK. Thanks. I'll make sure to do that. And yes, I don't see any agent associated with the job. I have tried force triggering this pipeline but unfortunately this hasn't worked either.

To be honest with you this isn't an issue for us any more as this pipeline was just a prototype for testing purposes. But I'll continue trying to figure out what went wrong as it would be good to know.

Thanks for your support and I'll get back to you when I have those logs.

**Comment by omnidavesz** [comment]
Hello again,

This problem is now back and causing us an issue with several more of our pipelines. I've tried changing the logging level as you've described but I'm having issues when attempting to restart the server.

Perhaps if I describe how we have GoCD set up it may help.

We run the Go server on a Docker container in an AWS ECS cluster. We also run the Go Agents in containers on a separate cluster in AWS ECS.

The steps I took were as follows:

1. SSH onto the EC2 instance hosting the Docker container with the Go server.
2. Open an interactive shell in the Docker container.
3. Used vim to set `log4j.logger.com.thoughtworks.go=DEBUG`
4. Attempted to run `/etc/init.d/go-server restart` in order to restart the server
5. Received java.io.FileNotFoundException for multiple files which definitely exist e.g. `Error trying to validate file log4j.properties: java.io.FileNotFoundException: /etc/go/log4j.properties (Permission denied)`
6. Attempted to stop the server using `/etc/init.d/go-server stop` but received “Go is not running” even though the server is definitely running.

Do you have any ideas as I'm not too sure what I should be doing next?

Regards,
David

**Comment by arvindsv** [comment]
Did you explore permission issues, as evidenced by `java.io.FileNotFoundException: /etc/go/log4j.properties (Permission denied)`? Do you use docker volumes? They've been known to be problematic with regards to permissions, because of the layers and aufs.

You mentioned not much in the logs. You should probably check that the server has permissions (user should be "go", etc) to write to `/var/log/go-server/go-server.log`.

**Comment by omnidavesz** [comment]
Hi,

I'm getting some details out but I don't think the log level is detailed enough i.e. I'm only getting out INFO and WARN level logs.

We're using a docker volume to store GoCD artifacts but I don't see how that would affect the logging and only the pipelines which use our lambdaDeploy template (which you can see in the zip file I sent above) fail.

I've not had a look at the permission issues but I'm not sure how to explore them. I've modified the `usr/share/go-server/server.sh` script with `set -x` at the top to see where the issue arises.

The issues occur after this line (the XXXXXXX bits are redacted data):

`exec /usr/lib/jvm/java-8-oracle/bin/java -server -Xms512m -Xmx1024m -XX:PermSize=128m -XX:MaxPermSize=256m -Dgo.database.provider=com.XXXXXXXXX.gocd.db.mysql.MySQLDatabase -Ddb.host=gocd-server-framework-goserverdbcluster-hwzqvavxum2d.cluster-XXXXXXXXXX.us-east-1.rds.amazonaws.com -Ddb.port=3306 -Ddb.name=goserverdb -Ddb.user=gouser -Ddb.password=XXXXXXX -Duser.language=en -Djruby.rack.request.size.threshold.bytes=30000000 -Duser.country=US -Dcruise.config.dir=/etc/go -Dcruise.config.file=/etc/go/cruise-config.xml -Dcruise.server.port=8153 -Dcruise.server.ssl.port=8154 -Djava.security.egd=file:/dev/./urandom -jar /usr/share/go-server/go.jar`

How would I go about checking the permissions?

**Comment by arvindsv** [comment]
> How would I go about checking the permissions?

I just meant checking to see why the go-server process cannot access /etc/go/log4j.properties. Something like this should do:

```
ls -l /etc/go /etc/go/log4j.properties
```

If you can share the logs, it might be more useful. We might be able to see things that you think are irrelevant.

**Comment by omnidavesz** [comment]
Oh OK.

So when I run the command you supplied I get this:

`-rw------- 1 root root 1891 Mar 14 17:32 /etc/go/log4j.properties

/etc/go:
total 380
-rw-r--r-- 1 root root   2482 Mar  6 14:44 agentkeystore
-rw-r--r-- 1 root root     16 Mar  6 14:44 cipher
-rw-r--r-- 1 root root    992 Mar  6 14:44 config.properties
-rw-r--r-- 1 root root 303224 Mar 15 08:53 cruise-config.xml
-rw-r--r-- 1 root root  44776 Mar  6 14:44 cruise-config.xsd
-rw-r--r-- 1 root root    607 Mar  6 14:45 gadget_truststore.jks
-rw-r--r-- 1 root root   3849 Mar  6 14:44 jetty.xml
-rw-r--r-- 1 root root   2669 Mar  6 14:44 jetty6.xml
-rw-r--r-- 1 root root   1482 Mar  6 14:44 keystore
-rw------- 1 root root   1891 Mar 14 17:32 log4j.properties
-rw-r--r-- 1 root root   2452 Mar  6 14:44 truststore`

I've had a look through the server logs for where I've kicked off my pipeline and found this:

`7382 2016-03-15 11:31:10,402  INFO [qtp1940447180-28] PipelineScheduler:134 - [Pipeline Schedule] [Requested] Manual trigger of pipeline 'PS_h-world-lambda-lambda-Java_DEV' requested by omnidavesz
   7383 2016-03-15 11:31:10,403  INFO [qtp1940447180-28] PipelineScheduler:140 - [Pipeline Schedule] [Accepted] Manual trigger of pipeline 'PS_h-world-lambda-lambda-Java_DEV' accepted for user omnidavesz
   7384 2016-03-15 11:31:10,405  INFO [qtp1940447180-28] PipelineScheduler:144 - [Pipeline Schedule] [Processed] Manual trigger of pipeline 'PS_h-world-lambda-lambda-Java_DEV' processed with result 'com.thoughtworks.go.serverhealth.ServerHealthState@14a73f61[healthState
   7384 Level=OK,type=<HealthStateType DATABASE_DISK_FULL LogScope[GLOBAL, scope=GLOBAL]>,message=,description=,expiryTime=<null>,timestamp=Tue Mar 15 11:31:10 UTC 2016]'
   7385 2016-03-15 11:31:12,223  INFO [consumeBuildCausesThread] ScheduleService:175 - [Pipeline Schedule] Scheduling pipeline PS_h-world-lambda-lambda-Java_DEV with build cause [ManualForcedBuildCause: Forced by David Szervanszky]
   7386 2016-03-15 11:31:12,264  INFO [consumeBuildCausesThread] PipelineRepository:88 - Start updating pipeline timeline
   7387 2016-03-15 11:31:12,266  INFO [consumeBuildCausesThread] PipelineRepository:92 - Pipeline timeline updated`

It looks like it's the DATABASE_DISK_FULL that's causing the issues. What database would that be referring to?

Regards,
David

**Comment by omnidavesz** [comment]
Found some more logs that look like they may have to do with the pipeline failures:

`INFO [consumeBuildCausesThread] ScheduleService:175 - [Pipeline Schedule] Scheduling pipeline PS_h-world-lambda-lambda-Java_DEV with build cause [ModificationBuildCause: modified by davidszervanszky <dszervanszky@omnifone.com>]
  14456 2016-03-15 12:31:49,261  INFO [consumeBuildCausesThread] PipelineRepository:88 - Start updating pipeline timeline
  14457 2016-03-15 12:31:49,263  INFO [consumeBuildCausesThread] PipelineRepository:92 - Pipeline timeline updated
  14458 2016-03-15 12:31:56,908  INFO [59@MessageListener for WorkFinder] Stage:218 - Stage is being completed by transition id: 34036
  14459 2016-03-15 12:31:57,067 ERROR [59@MessageListener for WorkFinder] StageService:235 - error notifying listener for stage <Stage id='5505' name='Setup_Stage' stageCounter='1' createdTime='2016-03-15 12:31:49.0' result='Failed'>
  14460 java.lang.RuntimeException: Multiple independent transactions are not permitted inside single transaction surrounding.
  14461         at com.thoughtworks.go.server.transaction.TransactionContext.transactionPushed(TransactionContext.java:47)
  14462         at com.thoughtworks.go.server.transaction.TransactionTemplate$2.doInTransaction(TransactionTemplate.java:37)
  14463         at org.springframework.transaction.support.TransactionTemplate.execute(TransactionTemplate.java:130)
  14464         at com.thoughtworks.go.server.transaction.TransactionTemplate.execute(TransactionTemplate.java:35)
  14465         at com.thoughtworks.go.server.dao.EnvironmentVariableSqlMapDao.load(EnvironmentVariableSqlMapDao.java:60)
  14466         at com.thoughtworks.go.server.dao.JobInstanceSqlMapDao.loadJobPlanAssociatedEntities(JobInstanceSqlMapDao.java:208)
  14467         at com.thoughtworks.go.server.dao.JobInstanceSqlMapDao.loadPlan(JobInstanceSqlMapDao.java:200)
  14468         at com.thoughtworks.go.server.service.JobInstanceService.loadOriginalJobPlan(JobInstanceService.java:207)
  14469         at com.thoughtworks.go.domain.XmlWriterContext.planFor(XmlWriterContext.java:62)
  14470         at com.thoughtworks.go.server.domain.xml.JobXmlViewModel.toXml(JobXmlViewModel.java:81)
  14471         at com.thoughtworks.go.server.service.XmlApiService.write(XmlApiService.java:49)
  14472         at com.thoughtworks.studios.shine.cruise.GoGRDDLResourceRDFizer.importURIUsingGRDDL(GoGRDDLResourceRDFizer.java:60)
  14473         at com.thoughtworks.studios.shine.cruise.stage.details.JobResourceImporter.importJob(JobResourceImporter.java:69)
  14474         at com.thoughtworks.studios.shine.cruise.stage.details.StageResourceImporter.importJobs(StageResourceImporter.java:124)
  14475         at com.thoughtworks.studios.shine.cruise.stage.details.StageResourceImporter.load(StageResourceImporter.java:84)
  14476         at com.thoughtworks.studios.shine.cruise.BackgroundStageLoader.handle(BackgroundStageLoader.java:71)
  14477         at com.thoughtworks.studios.shine.cruise.stage.feeds.StageAtomFeedsReader.readFromLatest(StageAtomFeedsReader.java:51)
  14478         at com.thoughtworks.studios.shine.cruise.BackgroundStageLoader.load(BackgroundStageLoader.java:61)
  14479         at com.thoughtworks.studios.shine.cruise.BackgroundStageLoaderStageStatusListener.stageStatusChanged(BackgroundStageLoader.java:104)
  14480         at com.thoughtworks.go.server.service.StageService$4.afterCommit(StageService.java:233)
  14481         at org.springframework.transaction.support.TransactionSynchronizationUtils.invokeAfterCommit(TransactionSynchronizationUtils.java:133)
  14482         at org.springframework.transaction.support.TransactionSynchronizationUtils.triggerAfterCommit(TransactionSynchronizationUtils.java:121)
  14483         at org.springframework.transaction.support.AbstractPlatformTransactionManager.triggerAfterCommit(AbstractPlatformTransactionManager.java:950)
  14484         at org.springframework.transaction.support.AbstractPlatformTransactionManager.processCommit(AbstractPlatformTransactionManager.java:796)
  14485         at org.springframework.transaction.support.AbstractPlatformTransactionManager.commit(AbstractPlatformTransactionManager.java:723)
  14486         at org.springframework.transaction.support.TransactionTemplate.execute(TransactionTemplate.java:147)
  14487         at com.thoughtworks.go.server.transaction.TransactionTemplate.execute(TransactionTemplate.java:35)
  14488         at com.thoughtworks.go.server.service.StageService.changeJob(StageService.java:468)
  14489         at com.thoughtworks.go.server.service.StageService.failJob(StageService.java:460)
  14490         at com.thoughtworks.go.server.service.ScheduleService.failJob(ScheduleService.java:693)
  14491         at com.thoughtworks.go.server.service.ScheduledPipelineLoader.pipelineWithPasswordAwareBuildCauseByBuildId(ScheduledPipelineLoader.java:78)
  14492         at com.thoughtworks.go.server.service.BuildAssignmentService$2.surrounding(BuildAssignmentService.java:188)
  14493         at com.thoughtworks.go.server.transaction.TransactionTemplate.transactionSurrounding(TransactionTemplate.java:71)
  14494         at com.thoughtworks.go.server.service.BuildAssignmentService.createWork(BuildAssignmentService.java:181)
  14495         at com.thoughtworks.go.server.service.BuildAssignmentService.assignWorkToAgent(BuildAssignmentService.java:106)
  14496         at com.thoughtworks.go.server.messaging.scheduling.BuildAssignmentServiceAssigner.assignWorkToAgent(BuildAssignmentServiceAssigner.java:44)
  14497         at com.thoughtworks.go.server.messaging.scheduling.WorkFinder.onMessage(WorkFinder.java:56)
  14498         at com.thoughtworks.go.server.messaging.scheduling.WorkFinder.onMessage(WorkFinder.java:30)
  14499         at com.thoughtworks.go.server.messaging.activemq.JMSMessageListenerAdapter.runImpl(JMSMessageListenerAdapter.java:65)
  14500         at com.thoughtworks.go.server.messaging.activemq.JMSMessageListenerAdapter.run(JMSMessageListenerAdapter.java:50)
  14501         at java.lang.Thread.run(Thread.java:745)
  14502 2016-03-15 12:31:57,068  INFO [59@MessageListener for WorkFinder] BuildAssignmentService:110 - [Agent Assignment] Assigned job [JobIdentifier[PS_h-world-lambda-lambda-Java_DEV, 31, 06ca59b790, Setup_Stage, 1, Setup_Job, 5716]] to agent [Agent [37312eca1256, 172.
  14502 17.0.3, d2d289c2-f844-4591-b339-3395227a6b7c]]`

**Comment by arvindsv** [comment]
> Multiple independent transactions are not permitted inside single transaction surrounding.

That could be the problem, I guess. /cc @jyotisingh might have some idea.

**Comment by omnidavesz** [comment]
Hi,

Would it be possible to get an update on this issue please?

I've also noticed that this only seems to be an issue for pipelines which we've migrated from one template to another.

Regards,
David

**Comment by omnidavesz** [comment]
Hello,

Sorry to keep chasing but this issue is holding up some of our key developments. Would it be possible to get an update on this issue or even just some information of how we can work around it?

Regards,
David

**Comment by omnidavesz** [comment]
Hey,

Are there any further details on this issue?

Regards,
David

**Comment by zabil** [comment]
Looks like it's bombing while trying to load environment variable related configuration. 
And from the stack trace, this fails to notify job completion listeners that copies console log files once a job is complete. That explains the empty console log files.

You must be able to see the console log files ```data/console``` directory under your go installation dir. 

To debug the original problem do you recall any changes around your environment variable configuration?

```
       <environmentvariables>
            <variable name="UPSTREAM">
                <value>#{UPSTREAM}</value>
            </variable>
            <variable name="UPSTAGE">
                <value>#{UPSTAGE}</value>
            </variable>
            <variable name="ARTIFACT_STAGE">
                <value>#{ARTIFACT_STAGE}</value>
            </variable>
            <variable name="ARTIFACT_JOB">
                <value>#{ARTIFACT_JOB}</value>
            </variable>
            <variable name="PREFIX">
                <value>#{PREFIX}</value>
            </variable>
            <variable name="ENV">
                <value>#{ENV}</value>
            </variable>
            <variable name="COMPONENT_NAME">
                <value>#{COMPONENT_NAME}</value>
            </variable>
            <variable name="RUNTIME">
                <value>#{RUNTIME}</value>
            </variable>
        </environmentvariables>
```

**Comment by omnidavesz** [comment]
Hi,

That's where I made the change to the template. I added the "RUNTIME" environment variable as part of the new template.

Regards,
David

**Comment by zabil** [comment]
Can you try removing ```RUNTIME``` from the template and see if you still have the problem?

**Comment by omnidavesz** [comment]
Hey,

I tried removing the env var from our Deploy template and got the same issue. I didn't however remove it from our Build template. If I do remove it from our build template the build will fail as we have a script which relies on that var being set.

Regards,
David

**Comment by omnidavesz** [comment]
Hey,

Actually, just to be perfectly clear I didn't change the original templates. I just made very similar ones with the new environment variable and changed the pipelines using the old templates to point at the new ones.

However I'm unable to go back to the previous templates as the scripts we use to build our projects rely on the RUNTIME variable to determine which packaging tool to use for our code.

Regards,
David

**Comment by zabil** [comment]
Aha. Now we've got some steps.

I assume you've modified the config xml pipeline tag to point to the new template by changing the value on the "template" attribute.

**Comment by omnidavesz** [comment]
I deleted the pipelines using the admin UI and then had the users regenerate the pipelines.

Which effectively did as you said in changing the pipeline tage to point to the new template.

**Comment by omnidavesz** [comment]
Hello,

Has there been any progress on this issue?

Regards,
David

**Comment by omnidavesz** [comment]
Hi,

It's been a week since I've last had an update on this. As I've mentioned in another thread I'm getting a lot of pressure on me about this and as we're still evaluating GoCD, these delays are giving my superiors cause for concern.

Would it be possible to get an update on the above issue?

Regards,
David

**Comment by zabil** [comment]
@omnidavesz we haven't been able to replicate the issue.

Can you take some time to replicate this on a new instance of go?

**Comment by arvindsv** [comment]
Unfortunately, on an open-source project, there are no guarantees such as this. If no one is updating it, then no one is working on it, and are probably busy with something else. I realize it's a problem for you, and I realize it's probably causing issues and that's understandable, but saying "It's been a week since I've last had an update on this" suggests an entitlement to an update, which does not exist in this model.

Having said all of that, I'd like you to try to set the environment variable `SHINE_ENABLED` to `false` on the Go Server. Here are [some details](https://docs.go.cd/current/advanced_usage/other_config_options.html) to do that. It should be enough to open /etc/default/go-server and add a line such as: `export SHINE_ENABLED=false` and restarting the server. Then, try and reproduce this issue.

**Comment by omnidavesz** [comment]
Hi there,

Thanks for the suggestion. I've attempted to implement it but unfortunately it hasn't worked.

However I was unable to follow your instructions to the letter. I noticed a few irregularities whilst attempting to follow them and I'm wondering whether the way we've deployed GoCD might account for the issues we're having.

Firstly we have the GoCD server deployed in a container on AWS. I made the change to the /etc/default/go-server file you recommended however I was unable to restart the server via the command line (I got a bunch of permission errors). Then I checked the status of the server and it was apparently stopped even though I was able to access the UI via my browser.

Therefore I modified the Dockerfile we used to build the server to include the `SHINE_ENABLED=false` config in the specified file and restarted the service. The config was definitely there but the issue persisted.

One final thing I tried was to upgrade the server (from 15.2 to 16.3) by changing the distribution being pulled in to the Dockerfile, rebuilding the Docker image and then deploying a new task with the updated image. However, when I logged onto the server via my browser the go version at the bottom was still specified as 15.2.

Regards,
David

**Comment by arvindsv** [comment]
These statements are a little scary:
  -  ... it was apparently stopped even though I was able to access the UI via my browser ...
  - Upgraded but "when I logged onto the server via my browser the go version at the bottom was still specified as 15.2"

Suggests a lot of inconsistent setup or not knowing where and how it is setup (docker container inside AWS). There are too many moving parts in here for me to even isolate this to a reproducible issue. I'll keep an eye out for this issue, and will think about it more, but since it is not happening to others, it suggests there's some environment setup issue.

**Comment by zabil** [comment]
@omnidavesz as @arvindsv has mentioned. 
There are a lot of things happening here. 
I've spent enough time trying to replicate and piece this this without relevant info. 

Really appreciate either you or someone in your team spend the time to replicate the original issue before asking for the next status update.

**Comment by omnidavesz** [comment]
@zabil and @arvindsv 

Thanks for looking into this. You're right that it's not worth looking into further until we understand what's happening on our end in terms of setup.

I'm going to be spending time investigating into how we even set GoCD up in the first place (unfortunately the engineers who did it have left).

If I discover any pertinant information I'll let you know.

Regards,
David

**Comment by davidedmonds** [comment]
Hi there,

I've managed to get what looks to be a similar issue on our Go.cd instance. The build in question has been around for a while and has previously worked, but is now instantly failing, generating the same stack trace mentioned above.

I've done a little bit of digging in that stack trace, and noticed that its trying to fail the job at that point (see https://github.com/gocd/gocd/blob/master/server/src/com/thoughtworks/go/server/service/ScheduledPipelineLoader.java#L79), and that the only reason that path would be used would be if one of the materials on the job was null.

Anyone have any idea what would cause a material to become null?

If it is helpful, we only have Git and Pipeline materials on that pipeline, it is based off a template, and I'm running Go Version: 16.4.0(3223-71ed3f0abf6ee673d1f0c7ca4e2262afae38dea6).

**Comment by davidedmonds** [comment]
Hmm. Having pushed another commit onto that project's branch, the build kicked off again successfully. So I suspect it'd be something to do with the Git materials at least.

**Comment by dlethin** [comment]
We are experiencing a similar issue -- We are trying to go back in our pipeline history and manually invoke the second stage of in a previous pipeline.  We see in the go-server.log file  that that the stage is being assigned.  However, after that the job is marked as failed.  There is no console output in the job detail tab, nor can we find any corresponding error message or stack traces in the go-server or go-agent log files.

this is what we see in our log file:

```
go-server.log:2019-05-31 16:15:13,113 INFO  [qtp1166807841-17] ScheduleService:230 - [Stage Schedule] Scheduling stage publish_x for pipeline myapp_release
go-server.log:2019-05-31 16:15:20,068 INFO  [101@MessageListener for WorkFinder] BuildAssignmentService:191 - [Agent Assignment] Assigned job [JobIdentifier[myapp_release, 84, 2.3.5.84, publish_x, 9, aws, 50810]] to agent [Agent [ip-10-100-14-193.mydomain, 10.100.14.193, 194988d9-53e3-4f57-9a21-e38c53fdbc9d]]
```

The other thing thing to note is that even though we see in the go-server.log file that the stage is being assigned,   on the detail page for the failed job we see this `AGENT:: Not yet assigned (Unknown)`.
There are two bits of information that might be playing a role here that are worth sharing:

- We have recently changed the underlying template for the pipeline. So the template that was active when the original pipeline first stage ran is no longer active.  However, the names of the stages and jobs within the pipeline remained the same.

- For one of the materials in the pipeline, the current definition points to a different git branch that it did when the pipeline run we are trying to run the second stage on.

In our case, these are our environment details:

###### Basic environment details

* Go Version: `19.4.0`
* JAVA Version: `1.8.0_101`
* OS: `Linux 3.10.0-327.28.2.el7.x86_64`

(We had noticed this problem on GoCD 19.3.0)

Not sure how to proceed in further troubleshooting this.  Any help would be greatly appreciated.

**Comment by arvindsv** [comment]
@dlethin I've opened a new issue (#6376) for your comment, since this issue is old and even though it looks similar, it might not be. Closing this one.
